cache/class-vary-cache.php

```diff
@@ -63,6 +63,10 @@ public static function remove_no_cache_for_user() {
 	public static function register_groups( $groups ) {
 		if ( is_array( $groups ) ) {
 			foreach ( $groups as $group ) {
+				if ( strpos( $group, self::GROUP_SEPARATOR ) !== false || strpos( $group, self::VALUE_SEPARATOR ) !== false ) {
+					trigger_error( 'Cannot use the group separator text in the group name', E_USER_WARNING );
```

Comment:
The error message should ideally include the separator:

```suggestion
					trigger_error( sprintf( 'Failed to register group; cannot use the delimiter values (`%s` or `%s`) in the group name', self::GROUP_SEPARATOR, self::VALUE_SEPARATOR ), E_USER_WARNING );
```